File: w06/d05/Raleigh/tictactoe/js/tic_tac_toe.js
function Player(color){
  // bonus feature : prompt for list of color choices
  this.color = color;
}
 
Player.prototype.makeMove = function(tileClicked) {
  // // make sure tile isn't already clicked; if already clicked, alert STOP THAT
  if (tileClicked.getAttribute("class") === "clicked") {
    return alert("That spot is already taken... move again.");
  } else {
    tileClicked.setAttribute("style", "background-color: " + this.color + ";");
    tileClicked.setAttribute("class", "clicked");
    play.counter++;
  };
}
 
function attachListener(tile){
  tile.addEventListener('click', function(){
    if(play.counter % 2 === 0){
      play.player1.makeMove(tile);
    } else {
      play.player2.makeMove(tile);
    }
  });
};
 
function createBoard(){
  var board = [];
 
  var tiles = document.querySelectorAll("td");
 
  _.each(tiles, function(tile){
    board.push(tile.getAttribute("id"));
  });
 
  _.each(tiles, function(tile){
    attachListener(tile);
  });
 
  return board;
}
 
function Game() {
  this.player1 = new Player("red");
  this.player2 = new Player("blue");
  this.board = createBoard();
  this.counter = 0;
}
 
window.onload = function(){
  window.play = new Game();
}

File: w07/d01/Julie/tictactoe.rb
class Game
  def initialize 
    @board = [[nil, nil, nil],[nil, nil, nil],[nil, nil, nil]]
    @counter = 0
    @player = "X"
    @win = false
  end


  def check_row
    @board.each do |i|
      if i[0] == i[1] && i[0] == i[2]
        @win = true
        puts "#{@player} won"
      else 
        nil
      end      

    end
  end

  def move(x, y)


  if @board[x][y].nil?
    if @player == "X"
       @board[x][y] = "X"
    else
       @board[x][y] = "circle"
    end
  else
    puts "You shall not pass!"
  end

    @player = ((@player == "X")? "circle" : "X")
    @counter += 1
  end


end



